src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SingleInitNullable.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Roslyn.Utilities;
+
+/// <summary>
+/// A lazily initialized version of <see cref="Nullable{T}"/> which uses the same space as a <see cref="Nullable{T}"/>.
+/// </summary>
+[NonCopyable]
+internal struct SingleInitNullable<T>
```

Comment:
💡 Can you relocate this to the compiler folder where **InterlockedOperations.cs** is defined, and then link it into this shared project? I'm trying to avoid having different initialization patterns available in all the layers.

Reply:
sure!